lib/javascript/JavascriptModulesPlugin.js

```diff
@@ -578,10 +578,17 @@ class JavascriptModulesPlugin {
 					const innerStrict = !allStrict && m.buildInfo.strict;
 					const iife = innerStrict || inlinedModules.size > 1 || chunkModules;
 					if (iife) {
-						source.add("!function() {\n");
-						if (innerStrict) source.add('"use strict";\n');
-						source.add(renderedModule);
-						source.add("\n}();\n");
+						if (runtimeTemplate.supportsArrowFunction()) {
```

Comment:
Check for `RuntimeGlobals.thisAsExports` when the module is using `this` (which actually point to the `exports` object in CommonJs)